org.eclipse.buildship.core.test/src/main/groovy/org/eclipse/buildship/core/internal/workspace/ImportingProjectCrossVersionTest.groovy

```diff
@@ -102,6 +103,42 @@ class ImportingProjectCrossVersionTest extends ProjectSynchronizationSpecificati
                 '''
             }
         }
+
+        simpleProjectDir = dir('simple-build') {
+            file 'settings.gradle', '''
+                rootProject.name = 'simple-build'
+            '''
+            file 'build.gradle', '''
+                apply plugin: 'java'
+                description = 'a simple project.'
+                task myTask {}
+            '''
+            dir('src/main/java/pkg') {
+                file 'Main.java', '''
+                    package pkg;
+
+                    public class Main {
+                        public static void main(String[] args) {
+                            System.out.println("Hello, world!");
+                        }
+                    }
+                '''
+            }
+        }
+    }
+
+    @Unroll
+    def "Can import a simple project build with Gradle #distribution.version"(GradleDistribution distribution) {
+        when:
+        importAndWait(simpleProjectDir, distribution)
+
+        then:
+        allProjects().size() == 1
+        numOfGradleErrorMarkers == 0
+        getPlatformLogErrors().size() == 0
+
+        where:
+        distribution << getSupportedGradleDistributions()
```

Comment:
`getSupportedGradleDistributions()` should be changed to only return Gradle distributions that support the Java version running the test (i.e. `eclipse.test.java.version`). 

This test IMHO should use `getAllGradleDistributions()` (to be implemented), and should assert that
- import with compatible java versions succeed
- import with incompatible java versions fail.